tt_eager/tensor/owned_buffer.hpp

```diff
@@ -34,6 +34,7 @@ struct Buffer {
 
     inline bool is_allocated() const{ return bool(this->shared_vector_); }
     inline const std::vector<T>& get() const { return *this->shared_vector_; }
+    inline const std::shared_ptr<std::vector<T>> get_ptr() const noexcept { return this->shared_vector_; }
```

Comment:
Why not use `data` method that already exists?

Reply:
Hey, one issue we ran into with Async mode was that the worker thread could not control the lifetime of a host buffer. This was due to raw ptrs (in this case with `buffer.data()`) being passed to `EnqueueWriteBuffer`. For owned buffers, we bypassed this by passing in a shared pointer, which we get through the new `get_ptr()`API.

If I try out your other suggestions, I likely don't need `get_ptr()` in the `borrowed_buffer` class.